openage/convert/singlefile.py

```diff
@@ -256,3 +259,79 @@ def read_smx_file(smx_path, main_palette, player_palette, output_path):
 
     # save as png
     tex.save(Directory(output_file.parent).root, output_file.name)
+
+
+def terrain_convert(drs, slp_path, palette_index, output_path, interfac=None):
+    """
+    Reads a SLP from a DRS archive and output a texture from it. It uses Cython code (for now pure Python), which is faster.
+    """
+    output_file = Path(output_path)
+
+    # open from drs archive
+    drs_file = DRS(drs)
+
+    info("opening slp in drs '%s:%s'...", drs.name, slp_path)
+    slp_file = drs_file.root[slp_path].open("rb")
+
+    if interfac:
+        # open the interface file if given
+        interfac_file = interfac
+
+    else:
+        # otherwise use the path of the drs.
+        interfac_file = Path(drs.name).with_name(
+            "interfac.drs").open("rb")  # pylint: disable=no-member
+
+    # open palette
+    info("opening palette in drs '%s:%s.bina'...",
+         interfac_file.name, palette_index)
+    palette_file = DRS(
+        interfac_file).root["%s.bina" % palette_index].open("rb")
+
+    info("parsing palette data...")
+    palette = ColorTable(palette_file.read())
+
+    # import here to prevent that the __main__ depends on SLP
+    # just by importing this singlefile.py.
+    from .slp import determine_rgba_matrix, SLP, SLPFrame, FrameInfo
```

Comment:
`determine_rgba_matrix` is a cdef C function and this means it is *not* callable from here because `singlefile.py`. You need to call `get_palette_info` for a frame as you do in line 321 as that will call `determine_rgba_matrix` from cython for you.